Source/WebKit/Shared/API/glib/WebKitContextMenu.cpp

```diff
@@ -60,6 +60,11 @@ struct _WebKitContextMenuPrivate {
     GList* items;
     WebKitContextMenuItem* parentItem;
     GRefPtr<GVariant> userData;
+#if PLATFORM(WPE)
+    gint positionX;
+    gint positionY;
+    gboolean hasPosition;
```

Comment:
I think those three could be replaced with an `std::optional<IntPoint>`:


```suggestion
    std::optional<IntPoint> position;
```

Then, if the position is `std::nullopt` that means the position is unset.

Reply:
I think we should follow an approach similar to GTK and have webkit_context_menu_get_event() that returns a WPEEVent that can be used to get the position.